pkg/traceql/ast.go

```diff
@@ -475,204 +476,406 @@ func (o UnaryOperation) referencesSpan() bool {
 // **********************
 // Statics
 // **********************
+
 type Static struct {
-	Type   StaticType
-	N      int
-	F      float64
-	S      string
-	B      bool
-	D      time.Duration
-	Status Status // todo: can we just use the N member for status and kind?
-	Kind   Kind
+	Type StaticType
+
+	valScalar  uint64   // used for int, float64, bool, time.Duration, Kind, and Status
+	valBytes   []byte   // used for string, []int, []float64, []bool
+	valStrings []string // used for []string
 }
 
-// nolint: revive
-func (Static) __fieldExpression() {}
+type StaticMapKey struct {
```

Comment:
would it be faster to do:

```
type StaticMapKey uintXX
```

and just hash all necessary fields? are we concerned about collisions?

Reply:
This is definitely an interesting option and I thought about implementing it like this. But I was concerned about collisions and @mdisibio was worried about this as well. I'm also not sure whether this solution is atually faster for strings because Go maps are highly optimized to work well with strings (they are probably also the most common use case for aggregations)